sklearn/linear_model/_ridge.py

```diff
@@ -137,7 +141,41 @@ def _mv(x):
     return coefs
 
 
-def _solve_lsqr(X, y, alpha, max_iter=None, tol=1e-3):
+def _solve_lsqr(
+    X,
+    y,
+    *,
+    alpha,
+    fit_intercept=True,
+    max_iter=None,
+    tol=1e-3,
+    X_offset=None,
+    X_scale=None,
+    sample_weight_sqrt=None,
+):
+    """Solve Ridge regression via LSQR.
+
+    We expect that y is always mean centered.
+    If X is dense, we expect it to be mean centered such that we can solve
+        ||y - Xw||_2^2 + alpha * ||w||_2^2
+
+    If X is sparse, we expect X_offset to be given such that we can solve
+        ||y - (X - X_offset)w||_2^2 + alpha * ||w||_2^2
+
+    With sample weights S=diag(sample_weight), this becomes
+        ||sqrt(S) (y - (X - X_offset) w)||_2^2 + alpha * ||w||_2^2
+    and we expect y and X to already be rescaled, i.e. sqrt(S) @ y, sqrt(S) @ X.
```

Comment:
Shall we be extra explicit or is the following more confusing?

```suggestion
    and we expect y and X to already be rescaled, i.e. sqrt(S) @ y, sqrt(S) @ X.
    In this case, X_offset is the sample-weighted mean of X before scaling
    by sqrt(X) than helpful.
```

Reply:
@lorentzenchr feel free to merge without accepting this suggestion if you feel it is not helping.

Reply:
Yes, let's be extra specific.